src/plugins/overlap.js

```diff
@@ -0,0 +1,86 @@
+                    scope.enabled = true;
+                }
+
+                if (scope.enabled){
+                    api.tasks.on.change(scope, function (task) {
+                        // on every task change check for overlaps
+                        scope.handleOverlaps(task);
+                    });
+
+                    // if overlaps flag changes this will update the css class.
+                    api.directives.on.new(scope, function(directiveName, taskScope, taskElement) {
+                        if (directiveName === 'ganttTask') {
+                            taskScope.$watch('task.model.overlaps',function(newOverlapStatus,oldOverlapStatus){
+                                if (newOverlapStatus!==oldOverlapStatus){
+                                   if(newOverlapStatus){
+                                       taskElement.addClass('gantt-task-overlaps');
+                                   } else {
+                                       taskElement.removeClass('gantt-task-overlaps');
+                                   }
+                                }
+                            });
+                        }
+                    });
+                }
+
+
+                scope.handleOverlaps = function (changedTask) {
+
+                    // Get all the tasks in the row.
+                    var allTasks = changedTask.row.model.tasks;
+
+                    // clear all overlaps flag.
+                    angular.forEach(allTasks,function(Task){
+                        Task.overlaps = false;
+                    });
+
+                    // set overlaps flag to each task that overlaps other task.
+                    angular.forEach(allTasks,function(currentTask){
+                        var currentStart,currentEnd;
+                        if (currentTask.from.isBefore(currentTask.to)){
+                            currentStart = currentTask.from;
+                            currentEnd = currentTask.to;
+                        } else {
+                            currentStart = currentTask.to;
+                            currentEnd = currentTask.from;
+                        }
+                        var currentRange = moment().range(currentStart, currentEnd);
+                        angular.forEach(allTasks,function(Task){
+                            if (currentTask.id !== Task.id){
+                                var Start,End;
+                                if (Task.from.isBefore(Task.to)){
+                                    Start = Task.from;
+                                    End = Task.to;
+                                } else {
+                                    Start = Task.to;
+                                    End = Task.from;
+                                }
+                                var Range = moment().range(Start, End);
+                                if (Range.overlaps(currentRange)){
+                                    Task.overlaps = currentTask.overlaps = true;
```

Comment:
It would be better to apply `gantt-task-overlaps` CSS class here, and remove the code with scope watcher from lines 25-37.

There can be many tasks in the gantt, and global watcher count is really performance critical.